### PR TITLE
PhpUnit 11 support

### PR DESCRIPTION
**Pull request by dmitryuk**
Closes https://github.com/martin-helmich/phpunit-json-assert/issues/54

### CODE REVIEW

**Comment by dmitryuk** [comment]
@martin-helmich please review MR

File: .gitattributes [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 .gitattributes export-ignore
 .codeclimate.yml export-ignore
+.github export-ignore
```
Comment (dmitryuk) [inline .gitattributes:5]: image

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -12,11 +12,15 @@ jobs:
...
-        phpunit: ['8.0', '9.0', '10.0']
+        php: ['8.0', '8.1', '8.2', '8.3']
+        phpunit: ['8.0', '9.0', '10.0', '11.0']
         exclude:
           - php: '8.0'
             phpunit: '10.0'
+          - php: '8.1'
+            phpunit: '11.0'
+          - php: '8.0'
+            phpunit: '11.0'
```
Comment (dmitryuk) [inline .github/workflows/php.yml:23]: PhpUnit supports only >=8.2

File: .github/workflows/php_coverage.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
     - name: Setup PHP
       uses: shivammathur/setup-php@v2
       with:
-        php-version: 8.0
+        php-version: 8.3
```
Comment (dmitryuk) [inline .github/workflows/php_coverage.yml:17]: Need update to fix code coverage

File: phpunit.xml [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
+      <report>
+        <html outputDirectory="build/coverage" lowUpperBound="35" highLowerBound="70"/>
+      </report>
     </coverage>
+    <source>
+      <include>
+        <directory suffix=".php">src</directory>
+      </include>
+    </source>
 </phpunit>
```
Comment (dmitryuk) [inline phpunit.xml:29]: Schema migrated

File: tests/Functional/JsonValueMatchesTest.php [hunk 5]
```diff
@@ -131,6 +135,7 @@ public function testJsonValueMatchesSucceedsWithAnyConstraint()
      * @param $expectedValue
      * @dataProvider dataForJsonValueEqualsCanFail
      */
+    #[DataProvider('dataForJsonValueEqualsCanFail')]
     public function testJsonValueMatchesCanFail($jsonDocument, $jsonPath, $expectedValue)
```
Comment (dmitryuk) [inline tests/Functional/JsonValueMatchesTest.php:139]: `@dataProvider` is deprecated

**Comment by AlexSkrypnyk** [comment]
@martin-helmich 
Can we please get this merged an released. This is blocking some projects to be updated to PHPUnit 11. Thank you

**Comment by martin-helmich** [comment]
Thanks for the ping, and sorry for the delay on this one. 🙏   Let me just see how to get the CI pipelines up and running again, and we can get this merged!

**Review by martin-helmich — APPROVED** [review]

**Pull request merged**